### ISSUE TITLE
Upgrading from 8.1.7 to 8.2.0 broke grafana with render plugin error

### ISSUE THREAD

**Issue by rda0** [grafana/grafana#40187]
**What happened**:

The automatic update broke grafana this morning.

dpkg log:

```
2021-10-08 06:34:54 startup packages configure
2021-10-08 06:34:59 startup archives unpack
2021-10-08 06:35:00 upgrade grafana:amd64 8.1.7 8.2.0
2021-10-08 06:35:00 status half-configured grafana:amd64 8.1.7
2021-10-08 06:35:00 status unpacked grafana:amd64 8.1.7
2021-10-08 06:35:00 status half-installed grafana:amd64 8.1.7
2021-10-08 06:35:03 status triggers-pending systemd:amd64 241-7~deb10u8
2021-10-08 06:35:03 status unpacked grafana:amd64 8.2.0
2021-10-08 06:35:03 startup packages configure
2021-10-08 06:35:03 configure grafana:amd64 8.2.0 <none>
2021-10-08 06:35:03 status unpacked grafana:amd64 8.2.0
2021-10-08 06:35:03 status half-configured grafana:amd64 8.2.0
2021-10-08 06:35:04 status installed grafana:amd64 8.2.0
2021-10-08 06:35:04 trigproc systemd:amd64 241-7~deb10u8 <none>
2021-10-08 06:35:04 status half-configured systemd:amd64 241-7~deb10u8
2021-10-08 06:35:04 status installed systemd:amd64 241-7~deb10u8
2021-10-08 06:35:08 startup packages configure
```

grafana log:

```
Oct  8 08:39:57 phd-test-influx systemd[1]: Started Grafana instance.

==> grafana/grafana.log <==
t=2021-10-08T08:39:57+0200 lvl=warn msg="falling back to legacy setting of 'min_interval_seconds'; please use the configuration option in the `unified_alerting` section if Grafana 8 alerts are enabled." logger=settings
t=2021-10-08T08:39:57+0200 lvl=warn msg="falling back to legacy setting of 'min_interval_seconds'; please use the configuration option in the `unified_alerting` section if Grafana 8 alerts are enabled." logger=settings
t=2021-10-08T08:39:57+0200 lvl=info msg="Config loaded from" logger=settings file=/usr/share/grafana/conf/defaults.ini
t=2021-10-08T08:39:57+0200 lvl=info msg="Config loaded from" logger=settings file=/etc/grafana/grafana.ini
t=2021-10-08T08:39:57+0200 lvl=info msg="Config overridden from command line" logger=settings arg="default.paths.data=/var/lib/grafana"
t=2021-10-08T08:39:57+0200 lvl=info msg="Config overridden from command line" logger=settings arg="default.paths.logs=/var/log/grafana"
t=2021-10-08T08:39:57+0200 lvl=info msg="Config overridden from command line" logger=settings arg="default.paths.plugins=/var/lib/grafana/plugins"
t=2021-10-08T08:39:57+0200 lvl=info msg="Config overridden from command line" logger=settings arg="default.paths.provisioning=/etc/grafana/provisioning"
t=2021-10-08T08:39:57+0200 lvl=info msg="Path Home" logger=settings path=/usr/share/grafana
t=2021-10-08T08:39:57+0200 lvl=info msg="Path Data" logger=settings path=/var/lib/grafana
t=2021-10-08T08:39:57+0200 lvl=info msg="Path Logs" logger=settings path=/var/log/grafana
t=2021-10-08T08:39:57+0200 lvl=info msg="Path Plugins" logger=settings path=/var/lib/grafana/plugins
t=2021-10-08T08:39:57+0200 lvl=info msg="Path Provisioning" logger=settings path=/etc/grafana/provisioning
t=2021-10-08T08:39:57+0200 lvl=info msg="App mode production" logger=settings
t=2021-10-08T08:39:57+0200 lvl=info msg="Connecting to DB" logger=sqlstore dbtype=mysql

==> syslog <==
Oct  8 08:39:57 phd-test-influx grafana-server[21099]: t=2021-10-08T08:39:57+0200 lvl=warn msg="falling back to legacy setting of 'min_interval_seconds'; please use the configuration option in the `unified_alerting` section if Grafana 8 alerts are enabled." logger=settings
Oct  8 08:39:57 phd-test-influx grafana-server[21099]: t=2021-10-08T08:39:57+0200 lvl=warn msg="falling back to legacy setting of 'min_interval_seconds'; please use the configuration option in the `unified_alerting` section if Grafana 8 alerts are enabled." logger=settings
Oct  8 08:39:57 phd-test-influx grafana-server[21099]: t=2021-10-08T08:39:57+0200 lvl=info msg="Config loaded from" logger=settings file=/usr/share/grafana/conf/defaults.ini
Oct  8 08:39:57 phd-test-influx grafana-server[21099]: t=2021-10-08T08:39:57+0200 lvl=info msg="Config loaded from" logger=settings file=/etc/grafana/grafana.ini
Oct  8 08:39:57 phd-test-influx grafana-server[21099]: t=2021-10-08T08:39:57+0200 lvl=info msg="Config overridden from command line" logger=settings arg="default.paths.data=/var/lib/grafana"
Oct  8 08:39:57 phd-test-influx grafana-server[21099]: t=2021-10-08T08:39:57+0200 lvl=info msg="Config overridden from command line" logger=settings arg="default.paths.logs=/var/log/grafana"
Oct  8 08:39:57 phd-test-influx grafana-server[21099]: t=2021-10-08T08:39:57+0200 lvl=info msg="Config overridden from command line" logger=settings arg="default.paths.plugins=/var/lib/grafana/plugins"
Oct  8 08:39:57 phd-test-influx grafana-server[21099]: t=2021-10-08T08:39:57+0200 lvl=info msg="Config overridden from command line" logger=settings arg="default.paths.provisioning=/etc/grafana/provisioning"
Oct  8 08:39:57 phd-test-influx grafana-server[21099]: t=2021-10-08T08:39:57+0200 lvl=info msg="Path Home" logger=settings path=/usr/share/grafana
Oct  8 08:39:57 phd-test-influx grafana-server[21099]: t=2021-10-08T08:39:57+0200 lvl=info msg="Path Data" logger=settings path=/var/lib/grafana
Oct  8 08:39:57 phd-test-influx grafana-server[21099]: t=2021-10-08T08:39:57+0200 lvl=info msg="Path Logs" logger=settings path=/var/log/grafana
Oct  8 08:39:57 phd-test-influx grafana-server[21099]: t=2021-10-08T08:39:57+0200 lvl=info msg="Path Plugins" logger=settings path=/var/lib/grafana/plugins
Oct  8 08:39:57 phd-test-influx grafana-server[21099]: t=2021-10-08T08:39:57+0200 lvl=info msg="Path Provisioning" logger=settings path=/etc/grafana/provisioning
Oct  8 08:39:57 phd-test-influx grafana-server[21099]: t=2021-10-08T08:39:57+0200 lvl=info msg="App mode production" logger=settings
Oct  8 08:39:57 phd-test-influx grafana-server[21099]: t=2021-10-08T08:39:57+0200 lvl=info msg="Connecting to DB" logger=sqlstore dbtype=mysql

==> grafana/grafana.log <==
t=2021-10-08T08:39:57+0200 lvl=info msg="Starting DB migrations" logger=migrator

==> syslog <==
Oct  8 08:39:57 phd-test-influx grafana-server[21099]: t=2021-10-08T08:39:57+0200 lvl=info msg="Starting DB migrations" logger=migrator

==> grafana/grafana.log <==
t=2021-10-08T08:39:57+0200 lvl=info msg="migrations completed" logger=migrator performed=0 skipped=346 duration=955.344µs

==> syslog <==
Oct  8 08:39:57 phd-test-influx grafana-server[21099]: t=2021-10-08T08:39:57+0200 lvl=info msg="migrations completed" logger=migrator performed=0 skipped=346 duration=955.344µs
Oct  8 08:39:57 phd-test-influx grafana-server[21099]: t=2021-10-08T08:39:57+0200 lvl=info msg="Starting plugin search" logger=plugins

==> grafana/grafana.log <==
t=2021-10-08T08:39:57+0200 lvl=info msg="Starting plugin search" logger=plugins

==> syslog <==
Oct  8 08:39:57 phd-test-influx grafana-server[21099]: t=2021-10-08T08:39:57+0200 lvl=info msg="Registering plugin" logger=plugins id=input

==> grafana/grafana.log <==
t=2021-10-08T08:39:57+0200 lvl=info msg="Registering plugin" logger=plugins id=input
t=2021-10-08T08:39:59+0200 lvl=info msg="Registering plugin" logger=plugins id=natel-discrete-panel

==> syslog <==
Oct  8 08:39:59 phd-test-influx grafana-server[21099]: t=2021-10-08T08:39:59+0200 lvl=info msg="Registering plugin" logger=plugins id=natel-discrete-panel
Oct  8 08:39:59 phd-test-influx grafana-server[21099]: t=2021-10-08T08:39:59+0200 lvl=info msg="Registering plugin" logger=plugins id=natel-plotly-panel

==> grafana/grafana.log <==
t=2021-10-08T08:39:59+0200 lvl=info msg="Registering plugin" logger=plugins id=natel-plotly-panel

==> syslog <==
Oct  8 08:39:59 phd-test-influx grafana-server[21099]: t=2021-10-08T08:39:59+0200 lvl=info msg="Registering plugin" logger=plugins id=bessler-pictureit-panel

==> grafana/grafana.log <==
t=2021-10-08T08:39:59+0200 lvl=info msg="Registering plugin" logger=plugins id=bessler-pictureit-panel
t=2021-10-08T08:39:59+0200 lvl=info msg="Registering plugin" logger=plugins id=grafana-image-renderer

==> syslog <==
Oct  8 08:39:59 phd-test-influx grafana-server[21099]: t=2021-10-08T08:39:59+0200 lvl=info msg="Registering plugin" logger=plugins id=grafana-image-renderer
Oct  8 08:39:59 phd-test-influx grafana-server[21099]: t=2021-10-08T08:39:59+0200 lvl=info msg="Registering plugin" logger=plugins id=jdbranham-diagram-panel

==> grafana/grafana.log <==
t=2021-10-08T08:39:59+0200 lvl=info msg="Registering plugin" logger=plugins id=jdbranham-diagram-panel
t=2021-10-08T08:39:59+0200 lvl=info msg="Registering plugin" logger=plugins id=larona-epict-panel

==> syslog <==
Oct  8 08:39:59 phd-test-influx grafana-server[21099]: t=2021-10-08T08:39:59+0200 lvl=info msg="Registering plugin" logger=plugins id=larona-epict-panel
Oct  8 08:39:59 phd-test-influx grafana-server[21099]: t=2021-10-08T08:39:59+0200 lvl=warn msg="Running an unsigned plugin" logger=plugins pluginID=btplc-alarm-box-panel pluginDir=/var/lib/grafana/plugins/btplc-alarm-box-panel/dist

==> grafana/grafana.log <==
t=2021-10-08T08:39:59+0200 lvl=warn msg="Running an unsigned plugin" logger=plugins pluginID=btplc-alarm-box-panel pluginDir=/var/lib/grafana/plugins/btplc-alarm-box-panel/dist

==> syslog <==
Oct  8 08:39:59 phd-test-influx grafana-server[21099]: t=2021-10-08T08:39:59+0200 lvl=info msg="Registering plugin" logger=plugins id=btplc-alarm-box-panel

==> grafana/grafana.log <==
t=2021-10-08T08:39:59+0200 lvl=info msg="Registering plugin" logger=plugins id=btplc-alarm-box-panel

==> syslog <==
Oct  8 08:39:59 phd-test-influx grafana-server[21099]: t=2021-10-08T08:39:59+0200 lvl=info msg="Registering plugin" logger=plugins id=flant-statusmap-panel

==> grafana/grafana.log <==
t=2021-10-08T08:39:59+0200 lvl=info msg="Registering plugin" logger=plugins id=flant-statusmap-panel

==> syslog <==
Oct  8 08:39:59 phd-test-influx grafana-server[21099]: t=2021-10-08T08:39:59+0200 lvl=info msg="Registering plugin" logger=plugins id=pierosavi-imageit-panel

==> grafana/grafana.log <==
t=2021-10-08T08:39:59+0200 lvl=info msg="Registering plugin" logger=plugins id=pierosavi-imageit-panel

==> syslog <==
Oct  8 08:39:59 phd-test-influx grafana-server[21099]: t=2021-10-08T08:39:59+0200 lvl=info msg="Registering plugin" logger=plugins id=vonage-status-panel

==> grafana/grafana.log <==
t=2021-10-08T08:39:59+0200 lvl=info msg="Registering plugin" logger=plugins id=vonage-status-panel
t=2021-10-08T08:39:59+0200 lvl=info msg="Live Push Gateway initialization" logger=live.push_http

==> syslog <==
Oct  8 08:39:59 phd-test-influx grafana-server[21099]: t=2021-10-08T08:39:59+0200 lvl=info msg="Live Push Gateway initialization" logger=live.push_http

==> grafana/grafana.log <==
t=2021-10-08T08:39:59+0200 lvl=info msg="Writing PID file" logger=server path=/run/grafana/grafana-server.pid pid=21099

==> syslog <==
Oct  8 08:39:59 phd-test-influx grafana-server[21099]: t=2021-10-08T08:39:59+0200 lvl=info msg="Writing PID file" logger=server path=/run/grafana/grafana-server.pid pid=21099

==> grafana/grafana.log <==
t=2021-10-08T08:39:59+0200 lvl=info msg="HTTP Server Listen" logger=http.server address=127.0.0.1:3000 protocol=http subUrl= socket=

==> syslog <==
Oct  8 08:39:59 phd-test-influx grafana-server[21099]: t=2021-10-08T08:39:59+0200 lvl=info msg="HTTP Server Listen" logger=http.server address=127.0.0.1:3000 protocol=http subUrl= socket=
Oct  8 08:39:59 phd-test-influx kernel: [6639496.101494] traps: plugin_start_li[21111] trap invalid opcode ip:833825 sp:7fff67144788 error:0 in plugin_start_linux_amd64[81e000+12e0000]
Oct  8 08:39:59 phd-test-influx grafana-server[21099]: t=2021-10-08T08:39:59+0200 lvl=eror msg="Stopped background service *rendering.RenderingService" logger=server reason="Failed to start renderer plugin: Unrecognized remote plugin message: \n\nThis usually means that the plugin is either invalid or simply\nneeds to be recompiled to support the latest protocol."

==> grafana/grafana.log <==
t=2021-10-08T08:39:59+0200 lvl=eror msg="Stopped background service *rendering.RenderingService" logger=server reason="Failed to start renderer plugin: Unrecognized remote plugin message: \n\nThis usually means that the plugin is either invalid or simply\nneeds to be recompiled to support the latest protocol."

==> syslog <==
Oct  8 08:39:59 phd-test-influx grafana-server[21099]: t=2021-10-08T08:39:59+0200 lvl=warn msg="plugin failed to exit gracefully" logger=plugins.backend pluginId=grafana-image-renderer

==> grafana/grafana.log <==
t=2021-10-08T08:39:59+0200 lvl=warn msg="plugin failed to exit gracefully" logger=plugins.backend pluginId=grafana-image-renderer
t=2021-10-08T08:39:59+0200 lvl=eror msg="Server shutdown" logger=server error="*rendering.RenderingService run error: Failed to start renderer plugin: Unrecognized remote plugin message: \n\nThis usually means that the plugin is either invalid or simply\nneeds to be recompiled to support the latest protocol."

==> syslog <==
Oct  8 08:39:59 phd-test-influx grafana-server[21099]: t=2021-10-08T08:39:59+0200 lvl=eror msg="Server shutdown" logger=server error="*rendering.RenderingService run error: Failed to start renderer plugin: Unrecognized remote plugin message: \n\nThis usually means that the plugin is either invalid or simply\nneeds to be recompiled to support the latest protocol."
Oct  8 08:39:59 phd-test-influx grafana-server[21099]: *rendering.RenderingService run error: Failed to start renderer plugin: Unrecognized remote plugin message:
Oct  8 08:39:59 phd-test-influx grafana-server[21099]: This usually means that the plugin is either invalid or simply
Oct  8 08:39:59 phd-test-influx grafana-server[21099]: needs to be recompiled to support the latest protocol.
Oct  8 08:39:59 phd-test-influx systemd[1]: grafana-server.service: Main process exited, code=exited, status=1/FAILURE
Oct  8 08:39:59 phd-test-influx systemd[1]: grafana-server.service: Failed with result 'exit-code'.
Oct  8 08:40:00 phd-test-influx systemd[1]: grafana-server.service: Service RestartSec=100ms expired, scheduling restart.
Oct  8 08:40:00 phd-test-influx systemd[1]: grafana-server.service: Scheduled restart job, restart counter is at 11.
Oct  8 08:40:00 phd-test-influx systemd[1]: Stopped Grafana instance.
```

**What you expected to happen**:

**How to reproduce it (as minimally and precisely as possible)**:

Upgrade grafana to 8.2.0

**Anything else we need to know?**:

Seems to be caused by `grafana-image-renderer`. Updating the plugin does not help:

```
+phd-influx:~# grafana-cli plugins upgrade-all
Updating grafana-image-renderer
Removing plugin: grafana-image-renderer
installing grafana-image-renderer @ 3.2.1
from: https://grafana.com/api/plugins/grafana-image-renderer/versions/3.2.1/download
into: /var/lib/grafana/plugins

✔ Installed grafana-image-renderer successfully
Please restart Grafana after installing plugins. Refer to Grafana documentation for instructions if necessary.
```

Removing the plugin seems to fix the problem:

```
+phd-influx:~# grafana-cli plugins remove grafana-image-renderer
Removing plugin: grafana-image-renderer
Please restart Grafana after installing plugins. Refer to Grafana documentation for instructions if necessary.

+phd-influx:~# systemctl restart grafana-server.service
```

Plugin list afterwards:

```
+phd-influx:~# grafana-cli plugins ls
installed plugins:
bessler-pictureit-panel @ 1.0.1
btplc-alarm-box-panel @ 1.0.8
flant-statusmap-panel @ 0.4.1
jdbranham-diagram-panel @ 1.7.3
larona-epict-panel @ 2.0.4
natel-discrete-panel @ 0.1.1
natel-plotly-panel @ 0.0.7
pierosavi-imageit-panel @ 1.0.5
vonage-status-panel @ 1.0.11
Please restart Grafana after installing plugins. Refer to Grafana documentation for instructions if necessary.
```

**Environment**:
- Grafana version: 8.2.0
- Data source type & version: InfluxDB 1.8.7
- OS Grafana is installed on: Debian GNU/Linux 10 (buster)
- User OS & Browser: NA
- Grafana plugins: see above (grafana-image-renderer)
- Others:

**Comment by nsteinmetz** [comment]
Same as me in #40188 

For this part:

```
t=2021-10-08T08:39:57+0200 lvl=warn msg="falling back to legacy setting of 'min_interval_seconds'; please use the configuration option in the `unified_alerting` section if Grafana 8 alerts are enabled." logger=settings
t=2021-10-08T08:39:57+0200 lvl=warn msg="falling back to legacy setting of 'min_interval_seconds'; please use the configuration option in the `unified_alerting` section if Grafana 8 alerts are enabled." logger=settings
```

I get rid of it by syncing my grafana.ini file with latest sample ini file - cf https://github.com/grafana/grafana/blob/main/conf/sample.ini

**Comment by rda0** [comment]
> Same as me in #40188 or [grafana/grafana-image-renderer#298](https://github.com/grafana/grafana-image-renderer/issues/298)
> 
> For this part:
> 
> ```
> t=2021-10-08T08:39:57+0200 lvl=warn msg="falling back to legacy setting of 'min_interval_seconds'; please use the configuration option in the `unified_alerting` section if Grafana 8 alerts are enabled." logger=settings
> t=2021-10-08T08:39:57+0200 lvl=warn msg="falling back to legacy setting of 'min_interval_seconds'; please use the configuration option in the `unified_alerting` section if Grafana 8 alerts are enabled." logger=settings
> ```
> 
> I get rid of it by syncing my grafana.ini file with latest sample ini file - cf https://github.com/grafana/grafana/blob/main/conf/sample.ini

I do not have any alerting config options set. Both sections `[unified_alerting]` and `[alerting]` are absent.

**Comment by peterdk** [comment]
Same issue here, 3 ubuntu 20.04 servers with grafana on them, failing to start the grafana-server after update.

**Comment by log1-c** [comment]
Can confirm. Works again on CentOs 8 with Grafana 8.2.0 and Renderer 3.2.1 :)

**Comment by kminehart** [comment]
For now I am going to close this issue as it's resolved by this PR: https://github.com/grafana/grafana/pull/38109

and will be released in v8.2.1

**Comment by kminehart** [comment]
ah, good catch, thank you. You're right, my mistake! I'll edit my comment. :)
